collectd/src/main/java/com/microsoft/applicationinsights/collectd/ApplicationInsightsWriter.java

```diff
@@ -0,0 +1,283 @@
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.collectd;
+
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.Hashtable;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.collectd.internal.ApplicationInsightsWriterLogger;
+import com.microsoft.applicationinsights.collectd.internal.PluginExclusion;
+import com.microsoft.applicationinsights.collectd.internal.WriterConfiguration;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.internal.util.LocalStringsUtils;
+import com.microsoft.applicationinsights.telemetry.MetricTelemetry;
+import org.collectd.api.CollectdConfigInterface;
+import org.collectd.api.CollectdInitInterface;
+import org.collectd.api.CollectdShutdownInterface;
+import org.collectd.api.CollectdWriteInterface;
+import org.collectd.api.Collectd;
+import org.collectd.api.ValueList;
+import org.collectd.api.DataSet;
+import org.collectd.api.DataSource;
+import org.collectd.api.OConfigItem;
+
+/**
+ * Created by yonisha on 4/26/2015.
+ */
+public class ApplicationInsightsWriter implements
+        CollectdConfigInterface,
+        CollectdInitInterface,
+        CollectdWriteInterface,
+        CollectdShutdownInterface {
+
+    // region Consts
+
+    protected static final String COLLECTD_USER_AGENT = "CollectD-User-Agent";
+    protected static final String TELEMETRY_HOST_PROPERTY_NAME = "CollectD-Host";
+    private static final int SUCCESS_CODE = 0;
+    private static final int CONFIGURATION_PHASE_ERROR_CODE = 1;
+    private static final int INITIALIZATION_PHASE_ERROR_CODE = 2;
+    private static final String PLUGIN_NAME = "ApplicationInsightsWriter";
+    private static final String DEFAULT_AI_LOGGER_OUTPUT = "CONSOLE";
+
+    // endregion Consts
+
+    // region Members
+
+    private TelemetryConfiguration telemetryConfiguration;
+    private TelemetryClient telemetryClient;
+    private Dictionary<String, PluginExclusion> excludedPluginsDictionary;
```

Comment:
[Dictionary](http://docs.oracle.com/javase/6/docs/api/java/util/Dictionary.html) class is obsolete so unless it is required by CollectD framework you should use the Map interface instead.